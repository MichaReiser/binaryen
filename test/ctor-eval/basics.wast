(module
  (memory 256 256)
  (data (i32.const 10) "waka waka waka waka waka")
  (export "test1" $test1)
  (export "test2" $test2)
  (export "test3" $test3)
  (func $test1
    (drop (i32.const 0)) ;; no work at all, really
  )
  (func $test2
    (drop (i32.load (i32.const 12))) ;; a safe load
  )
  (func $test3
    (drop (i32.store (i32.const 12) (i32.const 115))) ;; a safe store, should alter memory
    (drop (i32.store16 (i32.const 20) (i32.const 31353)))
    (drop (i32.store8 (i32.const 23) (i32.const 120)))
  )
)
