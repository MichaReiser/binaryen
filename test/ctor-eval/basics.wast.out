(module
 (type $0 (func))
 (memory $0 256 256)
 (data (i32.const 10) "was\00\00\00aka yzkx waka waka")
 (export "test1" (func $test1))
 (export "test2" (func $test2))
 (export "test3" (func $test3))
 (func $test1 (type $0)
  (nop)
 )
 (func $test2 (type $0)
  (nop)
 )
 (func $test3 (type $0)
  (nop)
 )
)
